src/Agent.Worker/Build/FileContainerServer.cs

```diff
@@ -184,82 +184,87 @@ public class FileContainerServer
-                        failedFiles.Add(fileToUpload);
-                    }
-                    else
-                    {
-                        context.Debug(StringUtil.Loc("FileUploadFinish", fileToUpload, uploadTimer.ElapsedMilliseconds));
+                            // output detail upload trace for the file.
+                            ConcurrentQueue<string> logQueue;
+                            if (_fileUploadTraceLog.TryGetValue(itemPath, out logQueue))
+                            {
+                                context.Output(StringUtil.Loc("FileUploadDetailTrace", itemPath));
+                                string message;
+                                while (logQueue.TryDequeue(out message))
+                                {
+                                    context.Output(message);
+                                }
+                            }
 
-                        // debug detail upload trace for the file.
-                        ConcurrentQueue<string> logQueue;
-                        if (_fileUploadTraceLog.TryGetValue(itemPath, out logQueue))
+                            // tracking file that failed to upload.
+                            failedFiles.Add(fileToUpload);
+                        }
+                        else
                         {
-                            context.Debug($"Detail upload trace for file: {itemPath}");
-                            string message;
-                            while (logQueue.TryDequeue(out message))
+                            context.Debug(StringUtil.Loc("FileUploadFinish", fileToUpload, uploadTimer.ElapsedMilliseconds));
+
+                            // debug detail upload trace for the file.
+                            ConcurrentQueue<string> logQueue;
+                            if (_fileUploadTraceLog.TryGetValue(itemPath, out logQueue))
                             {
-                                context.Debug(message);
+                                context.Debug($"Detail upload trace for file: {itemPath}");
+                                string message;
+                                while (logQueue.TryDequeue(out message))
+                                {
+                                    context.Debug(message);
+                                }
                             }
                         }
-                    }
 
-                    if (response != null)
-                    {
-                        response.Dispose();
-                        response = null;
+                        if (response != null)
+                        {
+                            response.Dispose();
+                            response = null;
+                        }
                     }
-                }
 
-                Interlocked.Increment(ref _filesProcessed);
+                    Interlocked.Increment(ref _filesProcessed);
+                } catch(Exception ex) {
```

Comment:
i would just do 
```C#
catch
{
    context.Error($"Error '{ex.Message)}' processing file: '{fileToUpload}'");
    throw;
}
```

I would just use message, since .ToString() will contains the stack as well.
also, use Error instead of Debug, since debug require System.Debug=true, and most time this kind of error happen, you won't get repro easily.
Loc the message if you change to Error();

Reply:
Changing to 'Output' (error doesn't exist on the context)